### PR TITLE
Spike on switching images/placeholders on scroll

### PR DESCRIPTION
**Pull request by mkitt**
[#138191235](https://www.pivotaltracker.com/story/show/138191235)
[#138191277](https://www.pivotaltracker.com/story/show/138191277)

### CODE REVIEW

**Comment by mkitt** [comment]
@rynbyjn. Wanted to get a spike around what I was thinking about on the 🐕 🚶. It's kind of working in this state but would need a bunch more hashed out. Something to chat about and compare with #118 when we get there.

File: src/components/viewport/ScrollComponent.js [hunk 1]
```diff
@@ -59,31 +60,42 @@ function getScrollHeight() {
...
   scrollObjects.forEach((obj) => {
-    const scrollProperties = getScrollProperties()
-    const scrollAction = getScrollAction(scrollProperties)
     callMethod(obj, 'onScroll', scrollProperties)
     if (scrollAction) {
       callMethod(obj, scrollAction, scrollProperties)
     }
+    if (screen !== scrollProperties.scrollScreen) {
+      screen = scrollProperties.scrollScreen
+      callMethod(obj, 'onScrollScreenChange', scrollProperties)
```
Comment (rynbyjn) [inline src/components/viewport/ScrollComponent.js]: I dig this living in the scroll component 🤘

File: src/containers/PostContainer.js [hunk 2]
```diff
@@ -241,13 +244,31 @@ class PostContainer extends Component {
...
+      isOnScreen: false,
       showComments: false,
       showLovers: false,
       showReposters: false,
     }
   }
 
-  shouldComponentUpdate(nextProps) {
+  componentDidMount() {
+    this.activeScreen = Math.abs(Math.ceil(this.ref.offsetTop / this.props.innerHeight))
```
Comment (rynbyjn) [inline src/containers/PostContainer.js]: This still has the same problem I was seeing where you can't call `setState` from `componentDidMount` which wouldn't update the `isOnScreen` state. So, images wouldn't be initially visible until a scroll actually happened or new props were handed to this, right?

**Comment by mkitt** [comment]
@rynbyjn closing this based on our findings. There is some good stuff in here which might be useful down the road though.

**Pull request closed without merging**